src/System.Reflection.Emit.ILGeneration/tests/Generator.cs

```diff
@@ -0,0 +1,427 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+
+namespace TestLibrary
+{
+    public static class Generator
+    {
+        internal static Random m_rand = new Random();
+        internal static int? seed = null;
+
+        public static int? Seed
+        {
+            get
+            {
+                if (seed.HasValue)
+                {
+                    return seed.Value;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            set
+            {
+                if (!(seed.HasValue))
+                {
+                    seed = value;
+                    if (seed.HasValue)
+                    {
+                        m_rand = new Random(seed.Value);
+                    }
+                }
+            }
+        }
+
+        // returns a byte array of random data
+        public static void GetBytes(int new_seed, byte[] buffer)
+        {
+            Seed = new_seed;
```

Comment:
Even if Random itself was safe, these helpers are all changing the seed, which is instantiating and store a new Random instance, so concurrent access could easily have race conditions that result in a different seed / Random instance being used than was intended.